controllers/topology/internal/templateutil/doc.go

```diff
@@ -0,0 +1,18 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package templateutil provides utils to clean up templates during deletion of MachineDeployments and MachineSets.
+package templateutil
```

Comment:
Fine for me. I added the util suffix because that seemed to be a pattern in mdutil and noderefutil.  I agree it's not great having these specific template funcs in this package and the others outside of it. But moving them into the package also doesn't feel right.
@vincepri WDYT?

Reply:
Moved it into util.go